src/golang/lib/workflow/artifact/artifact.go

```diff
@@ -222,11 +221,18 @@ func (a *ArtifactImpl) updateArtifactResultAfterComputation(
 		)
 		if err != nil {
 			log.Errorf("Unable to read artifact result metadata from storage and unmarshal: %v", err)
```

Comment:
Curious, is this error ever expected in any cases? If it's a really fatal error we should probably also add a Tip to the execution state?

Reply:
And make it a system error or something.

Reply:
I can add a comment here